File: tests/ResultCall.cs
namespace Demo {
    public partial class TRetornoZoom {
    }
    
    public partial class Foo {
        public TRetornoZoom Response(string body_string) {
            TRetornoZoom result;
            result = new TRetornoZoom();
            result.definirMensagem("00", body_string);
            return result;
        }
    }
}

File: tests/ResultCall.pas
namespace Demo;

type
  TRetornoZoom = public class
  public
    method definirMensagem(a: String; b: String);
  end;

  Foo = public class
  public
    method Response(body_string: String): TRetornoZoom;
  end;

implementation

method Foo.Response(body_string: String): TRetornoZoom;
begin
  result := new TRetornoZoom;
  result.definirMensagem("00", body_string);
end;

end.
